### ISSUE TITLE
IDE0200 is misreported and its editorconfig setting affects IDE0039

### ISSUE THREAD

**Issue by cremor** [dotnet/roslyn#65504]
**Version Used**:
Visual Studio 17.4.1

**Steps to Reproduce**:

1. Create a new .NET 7 console app project
2. Add an .editorconfig file with the following content:
```
root = true

[*.cs]
csharp_style_prefer_local_over_anonymous_function = true:suggestion
csharp_style_prefer_method_group_conversion = true:suggestion
```
3. Add a new class with this method:
```cs
public static void Test()
{
    Action<object> action = obj => Console.WriteLine(obj);
}
```
4. Notice that both IDE0039 (Use local function) and IDE0200 (Lambda expression can be removed) are reported as expected.
5. Change the `csharp_style_prefer_local_over_anonymous_function` setting between `true` and `false` a few times.
6. Notice that each change is immediately reflected in the messages shown in the error list and in the code window (as expected).
7. Make sure that `csharp_style_prefer_local_over_anonymous_function` stays set to `true`.
8. Change the `csharp_style_prefer_method_group_conversion` setting between `true` and `false` a few times.
9. Notice that the messages shown in the error list and in the code window are **not** updated for the changes.
10. Make sure that `csharp_style_prefer_method_group_conversion` stays set to `false`.
11. Rebuild, reopen the class file and the whole solution until the IDE0200 message is gone. This sometimes takes multiple tries.
  It might also be needed that the class file is automatically opened when the solution is opened, at least that seems to increase the probability that IDE0200 isn't reported any more.
12. Notice that once IDE0200 is finally gone, IDE0039 isn't shown any more either. But IDE0039 should still be shown because it's setting is still enabled.
13. Switch to another file and back to the class file.
14. Notice that IDE0039 is shown again and IDE0200 isn't. So this is correct now.
15. Close and reopen the solution.
16. There is a high chance that IDE0200 is now shown again altough it shouldn't be.

**Diagnostic Id**:
IDE0200 

**Expected Behavior**:
The error list and the code window should always report messages/code fixes based on the current .editorconfig settings.

**Actual Behavior**:
The error list and the code window report messages/code fixes that do not match the current .editorconfig settings.

**Comment by CyrusNajmabadi** [comment]
I don't repro this.  Changing the settings results in the right set of analyzer reports.
